LibraryWebApp/Controllers/ReservationCartController.cs

```diff
@@ -44,7 +49,41 @@ public IActionResult Delete(int id)
             HttpContext.Session.Set(WC.SessionCart, reservationCartList);
             return RedirectToAction(nameof(Index));
         }
-
+        public async Task<IActionResult> CreateReservation()
+        {
+            List<ReservationCartEntry> reservationCartList = new List<ReservationCartEntry>();
+            if (HttpContext.Session.Get<IEnumerable<ReservationCartEntry>>(WC.SessionCart) != null
+                && HttpContext.Session.Get<IEnumerable<ReservationCartEntry>>(WC.SessionCart).Count() > 0)
+            {
+                reservationCartList = HttpContext.Session.Get<List<ReservationCartEntry>>(WC.SessionCart);
+            }
+            List<int> bookInCart = reservationCartList.Select(i => i.BookIdInCart).ToList();
+            var todayDate = DateTime.Now;
+            var userName = HttpContext.User.Identity.Name;
+            var user = await userManager.FindByNameAsync(userName);
+            Reservation reservation = new Reservation()
+            {
+                DateOfReservation = todayDate,
+                PlannedDateOfReturn = todayDate.AddMonths(+1),
+                IdClient= user.Id,
+            };
+            _db.Reservation.Add(reservation);
+            _db.SaveChanges();
+            var p = reservation.Id;
```

Comment:
you don't need this variable, but if you insist change its name to be more descriptive.